src/main/java/domain/RandomLinkGenerator.java

```diff
@@ -0,0 +1,13 @@
+package domain;
+
+import java.security.SecureRandom;
+
+public class RandomLinkGenerator implements LinkGenerator {
+
+    private static final SecureRandom SECURE_RANDOM = new SecureRandom();
+
+    @Override
+    public Link generate() {
+        return Link.from(SECURE_RANDOM.nextBoolean());
+    }
```

Comment:
```suggestion
    public static boolean generate() {
        return SECURE_RANDOM.nextBoolean();
    }
    // generate를 호출하는 곳에서
    Link.from(RandomLinkGenerator.generate())
```
이렇게 사용하는건 어떤가요? 클래스명은 RandomBooleanGenerator가 되어야겠네요

Reply:
저는 세 가지 이유로 현재 방식이 더 낫다고 생각합니다.

첫 번째는 현재 RandomLinkGenerator.generate()를 호출하는 곳이 LineGenerator뿐인데 Line을 만들 때 List 안에 Link를 하나하나 추가하는 것이 더 자연스럽다고 생각합니다.

두 번째는 Ladder는 Line으로 이루어져 있고 Line은 Link로 이루어져 있는데 LineGenerator가 Line을 생성하는 역할을 하고 있으므로 LinkGenerator에서 Link를 생성하고 있는 것이 직관적이었습니다.

세 번째 이유는 RandomBooleanGenerator는 굳이 따로 클래스를 만드는 것이 아닌 랜덤이 필요한 클래스에서 Random 객체를 생성해 nextBoolean()을 호출하면 될 것이라고 생각하기 때문입니다.

다만 제가 제시한 세 가지 이유 모두 프로그래밍적으로 어떤 이득이 있다는 근거는 아니라는 생각이 들긴 합니다. 그래서 왜 이런 코멘트를 남겨주셨는지 이유가 궁금합니다.

Reply:
제가 제안드린 이유는 RandomLinkGenerator는 Link만을 오직 생성하는 클래스라 나중에 확장할수 있는 가능성이 아예 없는것 같아서요
RandomGenerator라는 클래스를 만들고 static 메서드로 구현한다면 언제 어디서든 Random한 boolean값이 필요한 경우 사용할 수 있어 쉽게 확장할 수 있다고생각했어요.
생성자나 정적 팩토리 메서드등을 통해 생성할 수 있는 것을 Generator라는 클래스를 만들어 다른 클래스에서 관리하는 비용을 늘리는게 아닌지도 고민해보면 좋을것 같아요!

Reply:
LineGenerator는 사다리게임에서 사용하기 때문에 필요하지만 확장 가능성은 거의 없기에 그대로 두는 것이고 LinkGenerator는 Boolean을 반환하는 객체인데 Boolean은 다른 객체들에서 사용될 가능성이 있기 때문에 BooleanGenerator으로 만들고 Link가 필요한 객체에서 Link로 바꾸는 방법도 되게 괜찮은 것 같습니다. 피드백 반영해서 수정했습니다!!